src/components/CreateUser.js

```diff
@@ -133,156 +139,95 @@ class CreateUser extends React.Component {
       .then(({ user }) => {})
       .catch(error => {
         console.log(error);
-        this.setState({ waiting: false, errorMessage: error.message || "failed to create user" });
+        let newErr = error.message || "";
+        newErr = newErr.replace("email address", "username");
+        this.setState({ waiting: false, errorMessage: newErr || "failed to create user" });
       });
 
     this.setState({ password: "" });
   };
 
+  renderErrorMessage = (msg, addBreak) => {
+    if (msg)
+      return (
+        <span>
+          <div className="login-form-input-error">{msg}</div>
+        </span>
+      );
+
+    return addBreak ? <br /> : null;
+  };
+
+  updateUsername = username => this.setState({ username });
+  updatePassword = password => this.setState({ password });
+
   renderInputs = () => (
-    <div className="create-form-input-list" style={{ width: "460px" }}>
-      <div className="create-form-input-header">Username</div>
-      <input
-        className="create-form-input"
-        type="text"
-        name="username"
-        placeholder=""
-        value={this.state.username}
-        onChange={e => {
-          this.setState({ username: e.target.value });
-        }}
-      />
-      {this.state.usernameMessage ? (
-        <div style={{ color: "red", fontSize: "0.8em" }}>{this.state.usernameMessage}</div>
-      ) : (
-        <br />
-      )}
-      <div className="create-form-input-header">Password</div>
-      <input
-        className="create-form-input"
-        type="password"
-        name="password"
-        placeholder=""
-        value={this.state.password}
-        onChange={e => {
-          this.setState({ password: e.target.value });
-        }}
-      />
-      {this.state.passwordMessage ? (
-        <div style={{ color: "red", fontSize: "0.8em" }}>{this.state.passwordMessage}</div>
-      ) : (
-        <br />
-      )}
+    <div className="login-form-input-list">
+      <div>
+        <LoginInput
+          type={"username"}
+          data={this.state.username}
+          waiting={this.state.waiting}
+          onChange={this.updateUsername}
+        />
+        <LoginInput
+          type={"password"}
+          data={this.state.password}
+          waiting={this.state.waiting}
+          onChange={this.updatePassword}
+        />
+        {this.renderErrorMessage(this.state.passwordMessage)}
```

Comment:
@connorbo97 should we explicitly do `this.renderErrorMessage(this.state.passwordMessage, false)`?

Reply:
Upon further consideration, this is fine with the specialized scope of the function.